server/src/main/java/io/crate/execution/ddl/tables/AddColumnRequest.java

```diff
@@ -0,0 +1,286 @@
+    // Checks are merged into mapping without knowing column name as it's contained in expression itself
+    // and thus we can keep them separately which is not the case for other constraints.
+    private final List<StreamableCheckConstraint> checkConstraints = new ArrayList<>();
+
+    private final List<StreamableColumnInfo> columns = new ArrayList<>();
+
+    public AddColumnRequest(StreamInput in) throws IOException {
+        super(in);
+        relationName = new RelationName(in);
+
+        int count = in.readVInt();
+        for (int i = 0; i < count; i++) {
+            checkConstraints.add(StreamableCheckConstraint.readFrom(in));
+        }
+
+        count = in.readVInt();
+        for (int i = 0; i < count; i++) {
+            columns.add(StreamableColumnInfo.readFrom(in, ""));
+        }
+    }
+
+    /**
+     * For dynamic mapping updates.
+     * Fields which cannot be resolved using Reference are initialized with default or null values.
+     */
+    public AddColumnRequest(RelationName relationName,
+                            List<Reference> columnRefs) {
+        this.relationName = relationName;
+        this.columns.addAll(columnRefs.stream().map(this::refToStreamableColumnInfo).collect(Collectors.toList()));
+    }
+
+    private StreamableColumnInfo refToStreamableColumnInfo(Reference reference) {
+        String geoTree = null;
+        Map<String, Object> geoProperties = new HashMap<>();
+        if (reference instanceof GeoReference geoRef) {
+            geoTree = geoRef.geoTree();
+            geoProperties.put("precision", geoRef.precision());
+            geoProperties.put("tree_levels", geoRef.treeLevels());
+            geoProperties.put("distance_error_pct", geoRef.distanceErrorPct());
+        }
+        return new StreamableColumnInfo(
+            reference.column().name(),
+            reference.column().fqn(),
+            reference.valueType(),
+            reference.position(),
+            false, // A primary key can be specified only on ADD COLUMN
+            true, // not-null can be specified only on ADD COLUMN
+            true, // Use default, can be specified only on ADD COLUMN
+            reference.indexType(),
+            reference.valueType().id() == ArrayType.ID,
+            null, // INDEX USING FULLTEXT can be specified only on ADD COLUMN
+            null, // generated expression can be specified only on ADD COLUMN
+            reference.columnPolicy(),
+            geoTree,
+            geoProperties,
+            List.of(), // copyToTargets is used when INDEX is specified, only for ADD COLUMN
+            List.of() // there is no tree-like Ref hierarchy in target.valueType().valueForInsert(row.get(i)), it's a single map
+        );
+    }
+
+    /**
+     * For ADD COLUMN.
+     */
+    public AddColumnRequest(RelationName relationName,
+                            AnalyzedColumnDefinition colToAdd,
```

Comment:
I was kinda hoping we'd reduce the surface where `AnalyzedColumnDefinition` is used.

I guess it is fine to keep this as is, but in general I think we should aim to keep it isolated. Partly because it's mutable and therefore hard to reason about how it is getting changed if it is used across several layers.

Reply:
I started drafting removal of this -  tried a new constructor with 15 fields or a Builder - felt like both were too verbose since it's not only 15 Builder methods/ctor fields but also handling children.

Current one is probably not the best option either just less verbose. Better to have it clean from the beginning - so let's just accept verbosity and introduce `StremableColumnInfo.Builder`? then I can even revert static factory method `of(Reference)` which I pushed recently - other components will use builder relying on what they have.

Probably need some special children handling like
```
Builder {

list children;
Builder children(Builder anoteherOneChild)
}
```
and in places we call it traverse the tree and add nodes to builder.